engine/base.ftl

```diff
@@ -765,17 +773,74 @@ are added.
                         "Values" : [],
                         "Children" : [],
                         "SubObjects" : false,
-                        "PopulateMissingChildren" : true
+                        "PopulateMissingChildren" : true,
+                        "Reference" : {
+                            "Schema" : "attributeset",
+                            "Type" : ""
+                        }
                     }
                 ] +
                 normalisedAttributes ]
         [/#if]
     [/#if]
 
-    [#-- Determine the attribute values --]
-    [#local result = {} ]
+    [#-- If attribute value is defined as a reference, evaluate the --]
+    [#-- reference and use the result as the attribute's Children.  --]
+    [#-- Reference.Schema is the type of schema referenced          --]
+    [#-- (component, reference, attributeset [default]) whilst      --]
+    [#-- Reference.Type is the key to reference within the          --]
+    [#-- schema composite object structure.                         --]
+    [#local evaluatedRefAttributes = []]
     [#if normalisedAttributes?has_content]
         [#list normalisedAttributes as attribute]
+            [#if attribute.Reference.Type?has_content ]
+            
+                [#-- Determine composite object --]
+                [#switch attribute.Reference.Schema]
+                    [#case "attributeset" ]
+                        [#local attributeRefValue = attributeSetConfiguration[attribute.Reference.Type]!{} ]
+                        [#break]
+                    [#case "blueprint" ]
+                        [#local attributeRefValue = blueprintObject[attribute.Reference.Type]!{} ]
+                        [#break]
```

Comment:
Not sure I can see when you would ref a blueprint, that would become a bit of inception. Since anything inside of this composite object is most likely going to be in the blueprint.